app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  def new
+    @current_user = User.find(params[:user_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @current_user = User.find(params[:user_id])
+    @user_id = params[:user_id]
+    @post_id = params[:post_id]
+    comment = Comment.new(comment_params.merge(author_id: @current_user.id, post_id: @post_id))
+    if comment.valid?
+      comment.save
+      redirect_to user_post_path(@user_id, @post_id)
+    else
+      redirect_to new_user_post_comment_path(@current_user.id)
+    end
+  end
+
```

Comment:
- [ ] It will be great to implement this method like: 
```
def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
```
Remember you already have the current user don't need to query the database every time. Also chaining the queries is usually a good idea. Again I was not able to test this method because your application is throwing errors. Kindly make sure you havent misspelled some of the variables